python/paddle/distributed/parallel.py

```diff
@@ -0,0 +1,297 @@
+    if args.cluster_node_ips is not None and args.node_ip is None:
+        raise ValueError("please input current node ip, "
+                         "cannot only give `cluster_node_ips`.")
+    default_node_ip = os.environ.get("PADDLE_MASTER_IPADDR", None)
+    default_node_ip = "127.0.0.1" if default_node_ip is None else default_node_ip
+    if args.node_ip is None:
+        args.node_ip = default_node_ip
+    if args.cluster_node_ips is None:
+        args.cluster_node_ips = default_node_ip
+
+    # NOTE(chenweihang): Here should set `started_port` before
+    # `get_cluster_and_pod` and keep each process's started_port
+    # is same, see [ why need set default master info before run? ]
+    args.started_port = options.get('started_port', None)
+    if args.started_port is None:
+        default_port = os.environ.get("PADDLE_MASTER_PORT", None)
+        if default_port is None:
+            raise RuntimeError(
+                "Data parallel training start failed. If you start data parallel "
+                "training by `paddle.distributed.launch` module, Please ensure "
+                "that one of the following rules is met:\n"
+                "  1. Do not set `paddle.distributed.init_parallel_env` argument "
+                "`rank` or set it to be -1;\n"
+                "  2. Set `paddle.distributed.init_parallel_env` start port for "
+                "parallel training by `started_port=**`, e.g. started_port=6170."
+            )
+        args.started_port = int(default_port)
+
+    args.use_paddlecloud = options.get('use_paddlecloud', False)
+    args.print_config = options.get('print_config', True)
+
+    # set default `selected_gpus`
+    # TODO(chenweihang): if users gived number of `selected_gpus`
+    # is not equal to the spawn's nprocs, it will cause error, 
+    # and because we remove the `proc num` argument of 
+    # `init_parallel_env`, when above error occured, we do not 
+    # have a good way to check, so users are not recommended to 
+    # use this parameter, it is best to delete
+    args.selected_gpus = options.get('selected_gpus', None)
+    if args.selected_gpus is None:
+        args.selected_gpus = os.environ.get("PADDLE_CUDA_VISIBLE_DEVICES", None)
+        if args.selected_gpus is None:
+            raise ValueError(
+                "Data parallel training start failed. If you start data parallel "
+                "training by `paddle.distributed.launch` module, Please ensure "
+                "that one of the following rules is met:\n"
+                "  1. Do not set `paddle.distributed.init_parallel_env` argument "
+                "`rank` or set it to be -1;\n"
+                "  2. Set `paddle.distributed.init_parallel_env` selected gpus of "
+                "parallel training by `selected_gpus=**`, e.g. selected_gpus='0,1,2,3'."
+            )
+
+    # reuse code of launch.py
+    cluster, pod = get_cluster_and_pod(args)
+
+    # remove useless env vars
+    os.environ.pop("http_proxy", None)
```

Comment:
seems intrusive to remove user space environment variable

Reply:
1. here are keep consistent with `launch` method, I think there may be a problem here, so the `launch` is implemented like this
2. here only remove subprocess env, It will not modify the user's original configuration

Reply:
I guess it wouldn't matter much, quite uncommon that user needs to download something through proxy in a training subroutine.